Automattic/woocommerce-payments
# Add level3 data to update intention call

Fixes #1937 

#### Changes proposed in this Pull Request

This PR is to add the level3 data to the update intention call when appropriate. The update intention call is used when checking out with UPE to get the final order total and line items. I didn't feel it was necessary to include the level3 data when creating the intention initially since it will always be updated when checking out with UPE. 

This PR uses a lot of existing functionality to generate and make requests with the level3 data so the updates are straightforward. The one call out is that the `$account` property on the gateway needed to be set to `protected` to allow the UPE gateway access.



#### Testing instructions
- Enable UPE on the site by setting the `_wcpay_feature_upe` field to 1 in the `wp_options` table.
- Perform a checkout using UPE.
- Upon completing the checkout, the payment intent should contain the level3 data. This can be checked in a couple ways:
    - Go into the order page in the admin area and click on the payment intention link (pi_xxxx)
    - Add a `console.log( charge )`  above this [line](https://github.com/Automattic/woocommerce-payments/blob/426df1bf74cd331169e8cb25bd7c717ec8570e37/client/payment-details/payment-method/card/index.js#L96)
    - Confirm that the console log of the charge data contains the level3 data.

![Level3-console](https://user-images.githubusercontent.com/68524302/120532397-a29eb280-c3ad-11eb-9558-faef9a2a1fbe.png)

 - And/or: log into the Stripe test account, look up the payment intention or charge and expand the details to see the level3 data.

![Level3-stripe](https://user-images.githubusercontent.com/68524302/120532995-45573100-c3ae-11eb-8ee2-d6165e3478f3.png)


-------------------
